src/components/ui/calendar.tsx

```diff
@@ -1,81 +1,216 @@
 import * as React from "react";
-import { DayPicker } from "react-day-picker";
+import {
+  DayPicker,
+  useDayPicker,
+  useNavigation,
+  type CaptionLayout,
+  type DropdownProps,
+} from "react-day-picker";
 
 import { buttonVariants } from "@/components/ui/button";
 import { icons } from "@/components/ui/icons";
+import { ScrollArea } from "@/components/ui/scroll-area";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+} from "@/components/ui/select";
+
+import { setMonth } from "@/lib/config/date-fns";
 
 import { cn } from "@/lib/utils";
 
 export type CalendarProps = React.ComponentProps<typeof DayPicker>;
 
-const currentYear = new Date().getFullYear();
-
 function Calendar({
   className,
   classNames,
   showOutsideDays = true,
-  fromYear = currentYear - 15,
-  toYear = currentYear + 7,
+  fromYear,
+  toYear,
+  captionLayout,
   ...props
 }: CalendarProps) {
+  const layout: CaptionLayout = captionLayout || "dropdown-buttons";
```

Comment:
This is mostly just for retaining the existing `default` behaviour, since it previously *forced* all the calendars to use the `dropdown-button` layout.